infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackSpatialCache.kt

```diff
@@ -109,8 +109,11 @@ data class ContextCache(
         network
             .search(Geometries.rectangle(boundingBox.x.min, boundingBox.y.min, boundingBox.x.max, boundingBox.y.max))
             .groupBy { hit -> hit.value().locationTrackVersion to hit.value().alignmentVersion }
-            .filter { (_, hits) ->
-                hits.any { hit -> hit.value().segment.segmentPoints.any { point -> boundingBox.contains(point) } }
+            .filter { (versions, hits) ->
+                val layoutSegments = getAlignment(versions.second).segments
+                hits.any { hit ->
+                    layoutSegments[hit.value().segmentIndex].segmentPoints.any { point -> boundingBox.contains(point) }
+                }
```

Comment:
Tässä ja alempana oikeastaan sen ylemmän kommentin koodihaisun ilmentymää. `ContextCache` kutsuu itseään kakuksi, mutta sitten se kuitenkin hakee aina asioita sisäisesti sen sijasta että ne olisi sillä tiedossa. Toki muistinkäytön näkökulmasta todennäköisesti sitten järkevää, mutta silti jotenkin jännä

Reply:
Minun päässä asia on jopa juuri toisin päin: Tällä tavallahan LocationTrackSpatialCache juurikin pitää vastuunsa pienenä. getAlignment kyllä toimii aina normaalitapauksessa nopeasti, ja se, että se toimii, on alignmentDaon vastuulla. Ennemminkin tässä oli aiemmin juuri vikana, että tämä luokka yritti ylläpitää tietoa, johon sillä ei ollut asiaa; nyt se ymmärtää, että sen tarvitsee vaan muistaa viite, millä päästä tietoon käsiksi.